src/Razor/src/Microsoft.AspNetCore.Razor.VSCode/src/Formatting/FormattingHandler.ts

```diff
@@ -46,8 +46,24 @@ export class FormattingHandler {
                 virtualHtmlUri,
                 formattingParams.options);
 
+            if (textEdits === undefined) {
+                return this.emptyFormattingResponse;
+            }
+
+            const htmlDocText = razorDocument.htmlDocument.getContent();
+            const lineCount = this.countLines(htmlDocText);
             const serializableTextEdits = Array<SerializableTextEdit>();
-            for (const textEdit of textEdits) {
+            for (let textEdit of textEdits) {
+                // The below workaround is needed due to a bug on the HTML side where
+                // they'll sometimes send us an end position that exceeds the length
+                // of the document. Tracked by https://github.com/microsoft/vscode/issues/175298.
+                if (textEdit.range.end.line > lineCount) {
```

Comment:
I'm guessing `range.end.line` is 1-based, but thought I'd ask to confirm that this shouldn't be `>=`

Reply:
`range.end.line` is actually 0-based, but I made lineCount also be 0-based. Maybe I should rename it because I see how it could be unclear 😅